custom_components/localtuya/cover.py

```diff
@@ -81,72 +99,68 @@ def __init__(
 
     @property
     def is_closing(self):
         """Return if cover is closing."""
-        # self.update()
         state = self._state
-        # print('is_closing() : state [{}]'.format(state))
-        if state == "off":
-            return True
-        return False
+        return state == self._config[CONF_CLOSE_CMD]
+
+    @property
+    def is_open(self):
+        """Return if the cover is open or not."""
+        if self._config[CONF_POSITIONING_MODE] != CONF_MODE_POSITION:
+            return None
+        else:
+            return self._current_cover_position == 100
 
     @property
     def is_closed(self):
         """Return if the cover is closed or not."""
-        # _LOGGER.info("running is_closed from cover")
-        # self.update()
-        state = self._state
-        # print('is_closed() : state [{}]'.format(state))
-        if state == "off":
-            return False
-        if state == "on":
-            return True
-        return None
+        if self._config[CONF_POSITIONING_MODE] != CONF_MODE_POSITION:
+            return None
+        else:
+            return self._current_cover_position == 0
 
     def set_cover_position(self, **kwargs):
-        # _LOGGER.info("running set_cover_position from cover")
         """Move the cover to a specific position."""
-        newpos = float(kwargs["position"])
-        #        _LOGGER.info("Set new pos: %f", newpos)
-
-        currpos = self.current_cover_position
-        posdiff = abs(newpos - currpos)
-        #       25 sec corrisponde alla chiusura/apertura completa
-        mydelay = posdiff / 2.0
-        if newpos > currpos:
-            #            _LOGGER.info("Opening to %f: delay %f", newpos, mydelay )
-            self.open_cover()
-        else:
-            #            _LOGGER.info("Closing to %f: delay %f", newpos, mydelay )
-            self.close_cover()
-        sleep(mydelay)
-        self.stop_cover()
-        self._position = 50  # newpos
+        _LOGGER.debug("Setting cover position: %r", kwargs[ATTR_POSITION])
+        if self._config[CONF_POSITIONING_MODE] == CONF_MODE_FAKE:
+            newpos = float(kwargs[ATTR_POSITION])
+
+            currpos = self.current_cover_position
+            posdiff = abs(newpos - currpos)
+            mydelay = posdiff / 50.0 * self._config[CONF_SPAN_TIME]
+            if newpos > currpos:
+                _LOGGER.debug("Opening to %f: delay %f", newpos, mydelay)
+                self.open_cover()
+            else:
+                _LOGGER.debug("Closing to %f: delay %f", newpos, mydelay)
+                self.close_cover()
+            sleep(mydelay)
+            self.stop_cover()
+            self._current_cover_position = 50
+            _LOGGER.debug("Done")
+
+        elif self._config[CONF_POSITIONING_MODE] == CONF_MODE_POSITION:
+            converted_position = int(kwargs[ATTR_POSITION])
+            if converted_position in range(0, 101) and self.has_config(CONF_SETPOS):
```

Comment:
Use `0 < converted_position <= 100` instead, seems excessive to look for an integer in a list (even if it's a generator in this case).

Reply:
Oooh, you are super right, I just copied it from the original PR and my knowledge of python was not enough to notice that this equals to a search in a list... 😮